subsys/bluetooth/services/bas/bas_bls.c

```diff
@@ -194,6 +194,23 @@ void bt_bas_bls_set_battery_charge_level(enum bt_bas_bls_battery_charge_level le
 	bls.power_state &= ~BATTERY_CHARGE_LEVEL_MASK;
 	bls.power_state |= (level << BATTERY_CHARGE_LEVEL_SHIFT) & BATTERY_CHARGE_LEVEL_MASK;
 	bt_bas_bls_update_battery_level_status();
+
+	/* Return if CONFIG_BT_BAS_BCS is not enabled. */
+	if (!IS_ENABLED(CONFIG_BT_BAS_BCS)) {
+		return;
+	}
+
+	/*
+	 * Set the BCS Critical Power State bit as per BAS 1.1 specification section 3.4.1.1:
+	 * The BCS Critical Power State bit should be set to true if the Battery Charge Level
+	 * is set to Critical in the Power State field.
+	 */
+	if (level == BT_BAS_BLS_CHARGE_LEVEL_CRITICAL) {
+		bt_bas_bcs_set_battery_critical_state(true);
+		return;
+	}
+
+	bt_bas_bcs_set_battery_critical_state(false);
```

Comment:
```suggestion
	/* Return if CONFIG_BT_BAS_BCS is not enabled. */
	if (IS_ENABLED(CONFIG_BT_BAS_BCS)) {
		/*
		 * Set the BCS Critical Power State bit as per BAS 1.1 specification section 3.4.1.1:
		 * The BCS Critical Power State bit should be set to true if the Battery Charge Level
		 * is set to Critical in the Power State field.
		 */
		if (level == BT_BAS_BLS_CHARGE_LEVEL_CRITICAL) {
			bt_bas_bcs_set_battery_critical_state(true);
			return;
		}
	
		bt_bas_bcs_set_battery_critical_state(false);
	}
```

Suggest to use this design instead. It's much clearer that the code here is guarded, and will also be easier to expand with the other optional characteristics. 

Ditto for `bt_bas_bls_set_service_required`